src/core/preview/preview.tsx

```diff
@@ -150,19 +165,21 @@ export class UIPPreview extends UIPPlugin {
     this.$iframe.contentWindow?.document.close();
     this.$iframe.contentWindow?.document.write(html);
     this.$iframe.contentWindow?.document.close();
+    this.$iframe.hidden = false;
     this.$iframe.title = title;
-
-    setTimeout(() => this.$iframe.hidden = false, 100);
   }
 
   /** Updates preview content from the model state changes */
   @listen({event: 'uip:change', target: ($this: UIPPreview) => $this.$root})
-  protected _onRootStateChange(e?: UIPChangeEvent): void {
+  protected async _onRootStateChange(e?: UIPChangeEvent): Promise<void> {
     this.$container.style.minHeight = `${this.$inner.offsetHeight}px`;
-    this.update(e);
+    this.$$attr('loading', true);
+
+    await this.update(e);
 
-    afterNextRender(() => this.$container.style.minHeight = '0px');
-    skipOneRender(() => {
+    this.$container.style.minHeight = '0px';
```

Comment:
```suggestion
    this.$container.style.minHeight = '0';
```

Reply:
Added to https://github.com/exadel-inc/ui-playground/issues/705 as well (not sure if that redundant or was left due to animation problem)